src/Parsers/ParserSetQuery.cpp

```diff
@@ -109,6 +109,8 @@ bool ParserSetQuery::parseNameValuePair(SettingChange & change, IParser::Pos & p
         value = std::make_shared<ASTLiteral>(Field(static_cast<UInt64>(1)));
     else if (ParserKeyword("FALSE").ignore(pos, expected))
         value = std::make_shared<ASTLiteral>(Field(static_cast<UInt64>(0)));
+    else if (ParserKeyword("DEFAULT").ignore(pos, expected))
+        value = std::make_shared<ASTLiteral>(Field("DEFAULT"));
```

Comment:
It seems like with the current implementation `SET SETTING=DEFAULT` is equal to `SET SETTING='DEFAULT'`.

I did not find any settings with a String type where 'default' value would be relevant and at the same time not be the default value, but just to be sure and to avoid any collisions: maybe we should make an ASTLiteral from (for example) a field typed Null? Then we can just check it like `value.isNull()` in the `src/Core/Settings.cpp`.
```suggestion
        value = std::make_shared<ASTLiteral>(Field());
```

Reply:
For some reason I thought that by adding a null check we might miss parsing errors, but it seems that such errors would all be caught in the parser and never reach code in `src/Core/Settings.cpp`, so I agree that this is a good idea

Reply:
It seems like we have custom settings feature where we allow users to assign a Null value to their custom setting. Since there is no way to differentiate between ASTLiteral being an actual Null literal and 'alias' for the default value, this also does not seem like a good solution.

Reply:
Wouldn't it be better to go back to the previous version? The collision in the previous version can only be done if someone wants to make the variable equal to "default". This could probably only affect custom settings, so it seems less dangerous.

Reply:
We should not put DEFAULT to a Field.

Reply:
Although the NULL field is ok for now.